homeassistant/components/control4/light.py

```diff
@@ -0,0 +1,224 @@
+        name: str,
+        idx: int,
+        device_name: str,
+        device_manufacturer: str,
+        device_model: str,
+        device_id: int,
+        is_dimmer: bool,
+        light_transition_time: int,
+        light_cold_start_transition_time: int,
+    ):
+        """Initialize Control4 light entity."""
+        super().__init__(
+            hass,
+            entry,
+            coordinator,
+            name,
+            idx,
+            device_name,
+            device_manufacturer,
+            device_model,
+            device_id,
+        )
+        self._is_dimmer = is_dimmer
+        self._transition_time = light_transition_time
+        self._cold_start_transition_time = light_cold_start_transition_time
+        self._c4_light = None
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._c4_light = C4Light(self.director, self._idx)
+
+    @property
+    def is_on(self):
+        """Return whether this light is on or off."""
+        return bool(self._coordinator.data[self._idx]["value"] > 0)
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        if self._is_dimmer:
+            return self._coordinator.data[self._idx]["value"] * 2.55
+        return None
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        flags = 0
+        if self._is_dimmer:
+            flags |= SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+        return flags
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn the entity on."""
+        if self._is_dimmer:
+            if ATTR_TRANSITION in kwargs:
+                transition_length = kwargs[ATTR_TRANSITION] * 1000
+            elif self.brightness == 0:
+                transition_length = self._cold_start_transition_time * 1000
+            else:
+                transition_length = self._transition_time * 1000
+            if ATTR_BRIGHTNESS in kwargs:
+                brightness = (kwargs[ATTR_BRIGHTNESS] / 255) * 100
+            else:
+                brightness = 100
+            await self._c4_light.rampToLevel(brightness, transition_length)
+        else:
+            transition_length = 0
+            await self._c4_light.setLevel(100)
+        if transition_length == 0:
+            transition_length = 1000
+        delay_time = (transition_length / 1000) + 0.7
+        _LOGGER.debug("Delaying light update by %s seconds", delay_time)
+        await asyncio.sleep(delay_time)
+        await self._coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn the entity off."""
+        if self._is_dimmer:
+            if ATTR_TRANSITION in kwargs:
+                transition_length = kwargs[ATTR_TRANSITION] * 1000
+            else:
+                transition_length = self._cold_start_transition_time * 1000
```

Comment:
We shouldn't implement this at a platform level. This kind of feature should be added to the light integration so it can apply to any integration. It also doesn't make much sense that something called "start" is used for turn off.

Reply:
I agree - it seems that a default transition function could be implemented in the light profiles system https://github.com/home-assistant/core/blob/master/homeassistant/components/light/__init__.py#L143

Reply:
I even think that we have a PR open for that #36747

Reply:
Hey that's nice! I'll remove the default transition functionality + option flow from this PR then. In the future the light integration could be updated to support different "cold start/already on" transition times.